numpy/__init__.pyi

```diff
@@ -1060,3 +1068,148 @@ def compress(
     axis: Optional[int] = ...,
     out: Optional[ndarray] = ...,
 ) -> ndarray: ...
+@overload
+def clip(
+    a: _Number,
+    a_min: Optional[ArrayLike],
```

Comment:
`a_min` and `a_max` can't both be `None`, so you could get more specific here with even more overloads.

Reply:
Added in https://github.com/numpy/numpy/pull/16647/commits/4df3cd8fea437ca96f7f475f1a71c412e7ebaa8c.
mypy's error for assigning `NoReturn` are, unfortunately, not the most descriptive, but it will have to do.

Reply:
Can't you do better, by defining:
```
    a_min: ArrayLike,
    a_max: Optional[ArrayLike],
```
and an overload,
```
    a_min: None,
    a_max: ArrayLike,
```

Reply:
I actually hadn't considered that; good idea.
Implemented in https://github.com/numpy/numpy/pull/16647/commits/9e59f5745bebc92817c87bd4b57bf49fb8581f34.